Tests/UnitTests/Networking/HTTPClientTests.swift

```diff
@@ -285,6 +285,47 @@ final class HTTPClientTests: BaseHTTPClientTests<MockETagManager> {
         expect(header.value) == "false"
     }
 
+    func testRequestWithStorefrontPassesHeader() {
+        let headerName = "X-Storefront"
+        self.systemInfo.stubbedStorefront = MockStorefront(countryCode: "USA")
+
+        let header: Atomic<String?> = nil
+
+        stub(condition: hasHeaderNamed(headerName)) { request in
+            header.value = request.value(forHTTPHeaderField: headerName)
+            return .emptySuccessResponse()
+        }
+
+        let request = HTTPRequest(method: .post([:]), path: .mockPath)
+
+        waitUntil { completion in
+            self.client.perform(request) { (_: EmptyResponse) in completion() }
+        }
+
+        expect(header.value) == "USA"
+    }
+
+    func testRequestsWithoutStorefrontDoNotSendHeader() {
+        let headerName = "X-Storefront"
+        self.systemInfo.stubbedStorefront = nil
+
+        var headerPresent = false
```

Comment:
This can be a false positive if the value is never changed. Maybe use the same approach as the other test?

Reply:
@NachoSoto I copied this way from the `testRequestsWithoutStorefrontDoNotSendHeader` test 😅 I tried the same approach as the other new test that I wrote but the header was still _there_ but just had a `nil` value. 

I was trying to find a way to test that the header wasn't sent at all 🤔 

Can I change it to `var headerPresent = true` so that this will default fail?

Reply:
Pushed a commit that I _think_ is better 🤷‍♂️  And also fixed one test that could have resulted in a false positive

Reply:
>  I tried the same approach as the other new test that I wrote but the header was still there but just had a nil value.

Probably because of the `X-Storefrontt` typo 🙃 

This is fine too, but I think the other approach is more explicit? Starting with `headerPresent = true` can be confusing to future readers of this test.

Reply:
Fixed